### PR TITLE
Set Pods group path in user project

### PR DESCRIPTION
**Pull request by Whirlwind**
There is not the path for the `Pods` group in user project. This PR will set the path to `Pods/` folder.
When user set the `Pods` group path, the project will build failed, because the CP will not set the xcconfig path relative from the Pods group. see #6194.

On the other hand, the path for Pods group will set to the `Pods/` and if the Pods is a symlink, the xcconfig path will not changed.

This is a separate PR for #7781

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Set Pods group path in user project  
  [Whirlwind](https://github.com/Whirlwind)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spec/unit/installer/user_project_integrator/target_integrator/xcconfig_integrator_spec.rb [hunk 1]
```diff
@@ -28,6 +28,21 @@ module Pod
...
+      XCConfigIntegrator.integrate(@pod_bundle, [@target])
+      @project['Pods'].path.should.equal @pod_bundle.relative_pods_root_path.to_s
+    end
+
+    it 'sets the Pods xcconfig\'s location relative path from Pods group' do
+      XCConfigIntegrator.integrate(@pod_bundle, [@target])
+      group = @project['Pods']
+      group.files.each do |ref|
+        if ref.display_name.end_with?('.xcconfig')
+          ref.path.should.match /^Target Support Files/
```
Comment (segiddins) [inline spec/unit/installer/user_project_integrator/target_integrator/xcconfig_integrator_spec.rb]: `ref.path.should.start_with`

File: lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb [hunk 2]
```diff
@@ -42,20 +42,33 @@ def self.integrate(pod_bundle, targets)
...
           def self.set_target_xcconfig(pod_bundle, target, config)
-            path = pod_bundle.xcconfig_relative_path(config.name)
-            group = config.project['Pods'] || config.project.new_group('Pods')
-            file_ref = group.files.find { |f| f.path == path }
+            # Xcode root group's path is absolute, we must get the relative path of the sandbox to the user project
+            group_path = pod_bundle.relative_pods_root_path
+            group = config.project['Pods'] || config.project.new_group('Pods', group_path)
+
+            # support user custom paths of Pods group and xcconfigs files.
+            group_path = Pathname.new(group.path || '')
```
Comment (segiddins) [inline lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: this looks like a bunch of complicated logic, could you extract it out to its own method?

Comment (Whirlwind) [reply lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: done.

File: CHANGELOG.md [hunk 3]
```diff
@@ -41,6 +41,11 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Whirlwind](https://github.com/Whirlwind)
   [#7878](https://github.com/CocoaPods/CocoaPods/pull/7878)
 
+* Set Pods group path in user project
```
Comment (amorde) [inline CHANGELOG.md]: `Set the path of the Pods group in the Pods project`

File: lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb [hunk 4]
```diff
@@ -97,7 +95,7 @@ def self.print_override_warning(pod_bundle, target, config, key)
             ]
             message = "The `#{target.name} [#{config.name}]` " \
               "target overrides the `#{key}` build setting defined in " \
-              "`#{pod_bundle.xcconfig_relative_path(config.name)}'. " \
+              "`#{pod_bundle.relative_pods_root_path + pod_bundle.xcconfig_relative_path(config.name)}'. " \
```
Comment (amorde) [inline lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: could this could be pulled out into a function? since its repeated multiple times

File: lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb [hunk 5]
```diff
@@ -42,26 +42,24 @@ def self.integrate(pod_bundle, targets)
...
-            group = config.project['Pods'] || config.project.new_group('Pods')
-            file_ref = group.files.find { |f| f.path == path }
+            file_ref = create_xcconfig_ref(pod_bundle, config)
+
             existing = config.base_configuration_reference
 
             set_base_configuration_reference = ->() do
-              file_ref ||= group.new_file(path)
               config.base_configuration_reference = file_ref
             end
```
Comment (amorde) [inline lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: This block of code should probably be inlined now since it's one line - it's only called in 2 places below this.

File: lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb [hunk 6]
```diff
@@ -42,26 +42,24 @@ def self.integrate(pod_bundle, targets)
...
 
             if existing && existing != file_ref
               if existing.real_path.to_path.start_with?(pod_bundle.sandbox.root.to_path << '/')
                 set_base_configuration_reference.call
-              elsif !xcconfig_includes_target_xcconfig?(config.base_configuration_reference, path)
+              elsif !xcconfig_includes_target_xcconfig?(config.base_configuration_reference, file_ref.path)
```
Comment (amorde) [inline lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: nit: Could just do `path = file_ref.path` above rather than changing these

File: lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb [hunk 7]
```diff
@@ -139,6 +137,43 @@ def self.xcconfig_includes_target_xcconfig?(base_config_ref, target_config_path)
...
+            # Xcode root group's path is absolute, we must get the relative path of the sandbox to the user project
+            group_path = pod_bundle.relative_pods_root_path
+            group = config.project['Pods'] || config.project.new_group('Pods', group_path)
+
+            # support user custom paths of Pods group and xcconfigs files.
+            group_path = Pathname.new(group.path || '')
+            xcconfig_path = pod_bundle.relative_pods_root_path + pod_bundle.xcconfig_relative_path(config.name)
+            path = xcconfig_path.relative_path_from(group_path)
+
+            file_ref = group.files.find { |f| f.display_name == path.basename.to_s }
```
Comment (amorde) [inline lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb]: Could you do:

```ruby
filename = path.basename.to_s
file_ref = group.files.find { |f| f.display_name == filename }
```
`basename` is a bit expensive to call in a loop

File: CHANGELOG.md (hunk 3 above)
Comment (Whirlwind) [reply CHANGELOG.md]: Is it in the Pods project? not user project?

Comment (amorde) [reply CHANGELOG.md]: Ah, my bad, there's actually. One in the user project and one in the Pods project. What you have is fine 👍

**Comment by Whirlwind** [comment]
@amorde I did some changed as you said, but I modify the `xcodeproj`. Please have a look at https://github.com/CocoaPods/Xcodeproj/pull/586 first. Thanks.

**Comment by amorde** [comment]
Looks like there was an error from ruby gems, I restarted the build

File: Gemfile.lock [hunk 8]
```diff
@@ -24,22 +24,22 @@ GIT
 
 GIT
   remote: https://github.com/CocoaPods/Nanaimo.git
-  revision: aa4d0c50772c4392039e70a88bd9cc944064f0ad
+  revision: cdd75d07a93dd51680ad75c78a40d813e67bd968
```
Comment (amorde) [inline Gemfile.lock]: @Whirlwind Could you undo the changes to this file?

Comment (Whirlwind) [reply Gemfile.lock]: I upgrade the `xcodeproj` gem and it have a newer dependency for `nanaimo`( from 0.2.5 to 0.2.6), so the `nanaimo` is updated.

**Comment by Whirlwind** [comment]
@amorde The CI still has some wrong.

File: Gemfile.lock (hunk 8 above)
Comment (amorde) [reply Gemfile.lock]: Ah ok, makes sense

**Comment by amorde** [comment]
@DanToml any ideas on the CI failure?

**Comment by DanToml** [comment]
I have no idea cc// @segiddins might have advice on debugging bundler connectivity issues tho.

File: lib/cocoapods/target/aggregate_target.rb [hunk 9]
```diff
@@ -290,7 +290,7 @@ def podfile_dir_relative_path
     # @return [String] The path of the xcconfig file relative to the root of
     #         the user project.
     #
-    def xcconfig_relative_path(config_name)
+    def xcconfig_relative_srcroot(config_name)
```
Comment (segiddins) [inline lib/cocoapods/target/aggregate_target.rb]: why was this method renamed?

Comment (Whirlwind) [reply lib/cocoapods/target/aggregate_target.rb]: I remove the `xcconfig_relative_path` and add a `xcconfig_relative_pods_root`. But I found that the xcconfig should get the path which relative user project. So I revert the changes. But I want to make a different with `Pod Root`, I change the name to `xcconfig_relative_srcroot`.

Comment (amorde) [reply lib/cocoapods/target/aggregate_target.rb]: I think keeping the name and just updating the function doc would be OK - specifically the section on what the function returns

**Comment by Whirlwind** [comment]
@segiddins could you fix the CI?

**Comment by segiddins** [comment]
The Circle build is green on master, I've gone ahead and rebuilt it.

**Comment by segiddins** [comment]
@Whirlwind try to rebase master, I just fixed the circle build

File: CHANGELOG.md [hunk 10]
```diff
@@ -45,6 +45,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Whirlwind](https://github.com/Whirlwind)
   [#7878](https://github.com/CocoaPods/CocoaPods/pull/7878)
 
+* Set the path of the Pods group in the user project 
+  [Whirlwind](https://github.com/Whirlwind)
+  [#7886](https://github.com/CocoaPods/CocoaPods/pull/7886)
+  [#6194](https://github.com/CocoaPods/CocoaPods/issues/6194)
```
[the remark this answers is not in the document]
Comment (Whirlwind) [reply CHANGELOG.md]: done

**Comment by segiddins** [comment]
Thanks for bearing with us!
